e2e/qgb/qgb_network.go

```diff
@@ -380,7 +380,7 @@ func (network QGBNetwork) WaitForBlockWithCustomTimeout(
 		case <-ctx.Done():
 			cancel()
 			if errors.Is(ctx.Err(), context.DeadlineExceeded) {
-				return fmt.Errorf(" chain didn't reach height in time")
+				return fmt.Errorf("height=%d: %w", height, ErrHeightNotReached)
```

Comment:
[question] would this error be improved if it explicitly stated the desired height and the current height?

Reply:
Yes, good catch, definitely. There is a lot to improved with error handling. For now, I will specify better that the height is the `designed height` and maybe later we can have a PR that always keeps tracks of the reached  heights and logs them in case it failed to achieve them.